src/mapeo-project.js

```diff
@@ -1036,15 +1042,34 @@ export class MapeoProject extends TypedEmitter {
 
   /**
    * @param {Attachment} attachment
-   * @returns {Promise<null | BlobId>}
+   * @returns {Promise<null | BlobRef>}
    */
-  async #tryGetBlobId(attachment) {
+  async #tryGetAttachmentBlob(attachment) {
     // Audio must not have variants
     for (const variant of VARIANT_EXPORT_ORDER) {
-      const blobId = buildBlobId(attachment, variant)
-      const entry = await this.#blobStore.entry(blobId)
-      if (!entry) continue
-      return blobId
+      try {
+        const blobId = buildBlobId(attachment, variant)
+        const entry = await this.#blobStore.entry(blobId)
+        if (!entry) continue
+        const metadata = entry.value.metadata
+        if (!metadata || typeof metadata !== 'object') continue
+        if (
+          !('mimeType' in metadata && typeof metadata.mimeType === 'string')
+        ) {
+          this.#l.log('Blob missing mime type', blobId, entry)
+          continue
+        }
```

Comment:
I don't think we should not export attachments without metadata. This field is optional, and early versions of comapeo might not have metadata or mimetype, and I think it's best to lean towards getting all data out.